### PR TITLE
Make Hierarchy::Item#short unique per level

### PR DESCRIPTION
**Pull request by mereghost**
#### Relate WP: [OP#58852](https://community.openproject.org/projects/document-workflows-stream/work_packages/58852)

Gives `#short` the same treatment we have for labels.

### CODE REVIEW

**Review by apfohl — APPROVED** [review]

File: config/locales/en.yml [hunk 1]
```diff
@@ -299,8 +299,16 @@ en:
...
+        short:
+          not_unique: "must be unique within the same hierarchy level"
+        item:
+          root_item: "cannot be a root item"
+          not_persisted: "must be an already existing item"
     rules:
       label: "Label"
+      short: "Short"
+      item: "Item"
+      parent: "Item"
```
Comment (Kharonus) [inline config/locales/en.yml]: 🔴 This one will clash with my current PR. Why not translating key `parent` to `Parent`? And if it is a very specific translation (for this contract), then it should be below a namespace.

File: app/contracts/custom_fields/hierarchy/update_item_contract.rb [hunk 2]
```diff
@@ -31,23 +31,29 @@
...
+
       params do
         required(:item).filled(type?: CustomField::Hierarchy::Item)
         optional(:label).filled(:string)
         optional(:short).filled(:string)
       end
 
       rule(:item) do
-        key.failure("must exist") if value.new_record?
-        key.failure("must not be a root item") if value.root?
```
Comment (Kharonus) [inline app/contracts/custom_fields/hierarchy/update_item_contract.rb:42]: 🟢 nice

**Review by Kharonus — CHANGES_REQUESTED** [review]

File: config/locales/en.yml (hunk 1 above)
Comment (mereghost) [reply config/locales/en.yml]: I'm ok with that.

**Review by Kharonus — APPROVED** [review]
LGTM

**Pull request merged**